src/detectors/builtin/etaLikeSimplifications.ts

```diff
@@ -0,0 +1,190 @@
+    warnings: MistiTactWarning[],
+  ): void {
+    if (stmt.kind === "statement_condition") {
+      const ifStmt = stmt;
+      if (
+        ifStmt.trueStatements.length === 1 &&
+        ifStmt.falseStatements &&
+        ifStmt.falseStatements.length === 1 &&
+        ifStmt.trueStatements[0].kind === "statement_return" &&
+        ifStmt.falseStatements[0].kind === "statement_return"
+      ) {
+        const trueReturn = ifStmt.trueStatements[0] as AstStatementReturn;
+        const falseReturn = ifStmt.falseStatements[0] as AstStatementReturn;
+        if (
+          this.isBooleanLiteral(trueReturn.expression, true) &&
+          this.isBooleanLiteral(falseReturn.expression, false)
+        ) {
+          warnings.push(
+            this.makeWarning(
+              "Simplify 'if' statement by returning the condition directly",
+              stmt.loc,
+              {
+                suggestion: "Replace with 'return condition;'",
+              },
+            ),
+          );
+        }
+      }
+    }
+  }
+
+  private checkExpression(
+    expr: AstExpression,
+    warnings: MistiTactWarning[],
+  ): void {
+    if (expr.kind === "op_binary") {
+      const binaryExpr = expr as AstOpBinary;
+      if (binaryExpr.op === "==" || binaryExpr.op === "!=") {
+        const { right } = binaryExpr;
+        if (this.isBooleanLiteral(right)) {
+          warnings.push(
+            this.makeWarning(
+              "Redundant comparison with boolean literal",
+              expr.loc,
+              {
+                suggestion: `Use '${this.getSimplifiedBooleanExpression(
+                  binaryExpr,
+                )}' instead`,
+              },
+            ),
+          );
+        }
+      }
+    }
+    if (expr.kind === "conditional") {
+      const conditionalExpr = expr as AstConditional;
+      if (
+        this.isBooleanLiteral(conditionalExpr.thenBranch, true) &&
+        this.isBooleanLiteral(conditionalExpr.elseBranch, false)
+      ) {
+        warnings.push(
+          this.makeWarning(
+            "Simplify conditional expression by using the condition directly",
+            expr.loc,
+            {
+              suggestion: `Use '${this.getConditionText(
+                conditionalExpr.condition,
+              )}' instead`,
+            },
+          ),
+        );
+      }
+    }
+  }
+
+  private isBooleanLiteral(
+    expr: AstExpression | null | undefined,
+    value?: boolean,
+  ): boolean {
+    if (!expr) return false;
+    if (value === undefined) {
+      return evalToType(expr, "boolean") !== undefined;
+    }
+    return evalsToValue(expr, "boolean", value);
+  }
+
+  private getSimplifiedBooleanExpression(binaryExpr: AstOpBinary): string {
+    const exprText = (expr: AstExpression): string => {
+      if (expr.kind === "id") {
+        return idText(expr);
+      }
+      return "expression";
+    };
+
+    if (this.isBooleanLiteral(binaryExpr.right, true)) {
+      if (binaryExpr.op === "==") {
```

Comment:
I would simplify creating a constant here: `const negated = binaryExpr.op === "=="` and using it in this function.